craft_application/application.py

```diff
@@ -432,14 +432,18 @@ def _pre_run(self, dispatcher: craft_cli.Dispatcher) -> None:
         # Some commands might have a project_dir parameter. Those commands and
         # only those commands should get a project directory, but only when
         # not managed.
+        cmdline_project_dir = getattr(dispatcher.parsed_args(), "project_dir", None)
         if self.is_managed():
             self.project_dir = pathlib.Path("/root/project")
-        elif project_dir := getattr(dispatcher.parsed_args(), "project_dir", None):
-            self.project_dir = pathlib.Path(project_dir).expanduser().resolve()
+            if cmdline_project_dir:
+                # Should be emit.message?  Or some warning mechanism?
+                craft_cli.emit.verbose("In managed mode, ignoring --project-dir")
```

Comment:
```suggestion
```

While I generally appreciate more logging, in this case it's unnecessary. The manager always passes its full command line to the managed instance, so this is essentially logging that the user passed `--project-dir` in the outer instance. (Managed instances should never be called on their own.)

We tend not to use `emit.message` very much, and considering the outcome here (normal operation), it's at best `debug` (if something goes wrong we can see it), but more likely `trace` (if something went wrong about this, it's a pretty specialised debugging that's needed).

Reply:
Ok good to know thanks, I'll revert this one.